SoumyaSubhrajit/VehicleRentalHub
# adding test files

# Pull Request: Adding Jest Tests and Supertest

## Description

This pull request introduces comprehensive testing for the VehicleRentalHub project using Jest for unit tests and Supertest for endpoint testing. The goal is to ensure the reliability and stability of the application by covering critical functions and API endpoints.

## Changes Made

- Added Jest configuration (`jest.config.js`) for unit testing.
- Created unit tests for critical functions in the `src` directory.
- Integrated Supertest for API endpoint testing in the `tests` directory.

## Test Coverage

- **Unit Tests**: Implemented unit tests for functions related to database interactions, validation, and business logic.
- **API Endpoint Tests**: Utilized Supertest to conduct endpoint testing for key API routes, including `GET /api/vehicles` and `POST /api/bookings`.

## How to Test

1. Clone this branch locally.
2. Run `npm install` to install dependencies.
3. Execute `npm test` to run Jest tests.
4. Check the test results and coverage.

## Additional Notes

- Ensure that the application is properly configured for testing environments.
- Review the test files for documentation and comments explaining test cases.


Include any relevant screenshots or visuals
